### PR TITLE
Write system log during upgrade.

### PR DESCRIPTION
**Pull request by r0mant**
This PR makes sure that `telekube-system` log file is written during automatic upgrade on the node where master agent is running. Also includes the following changes/improvements:

* Capture update hook output and write it to the system log as well, similar to what install does.
* Move default location of our log files to `/var/log` instead of the "current working dir". This gives us a stable well-known location where these files can be found, as opposed to currently having to search for them everywhere because the working dir may be completely different depending on how the command was run (e.g. if it's run via an agent, then the file will end up in / and so on).
* Do not truncate telekube-system log file every time. I think it doesn't make sense to truncate it since it oftentimes useful to look at the past logs.

Closes https://github.com/gravitational/gravity.e/issues/3596.

### CODE REVIEW

File: lib/update/phase_app.go [hunk 1]
```diff
@@ -190,11 +192,25 @@ func (p *phaseApp) runHooks(ctx context.Context, hooks ...schema.HookType) error
...
+			for scanner.Scan() {
+				log.WithField(trace.Component, "hook").Info(scanner.Text())
+			}
+			if err := scanner.Err(); err != nil {
+				log.Warnf("Failed to stream hook logs: %v.", err)
+			}
+		}()
+		_, err = app.StreamAppHook(ctx, p.Apps, req, writer)
+		if err != nil {
+			return trace.Wrap(err, "%v %s hook failed", p.Package, hook)
```
Comment (a-palchikov) [inline lib/update/phase_app.go]: `writer.Close` missing

File: lib/update/phase_app.go [hunk 2]
```diff
@@ -190,11 +192,25 @@ func (p *phaseApp) runHooks(ctx context.Context, hooks ...schema.HookType) error
 			}
 			return trace.Wrap(err)
 		}
-		_, out, err := app.RunAppHook(ctx, p.Apps, req)
+		reader, writer := io.Pipe()
+		go func() {
+			defer reader.Close()
+			scanner := bufio.NewScanner(reader)
+			for scanner.Scan() {
+				log.WithField(trace.Component, "hook").Info(scanner.Text())
```
Comment (a-palchikov) [inline lib/update/phase_app.go]: I think the streaming part should be factored out and use the name of the hook in the log at the very least:
```go
func (p *phaseApp) runHooks(ctx context.Context, hooks ...schema.HookType) error {
	// ...
	reader, writer := io.Pipe()
	go streamHook(hook, reader)
	// ...
}

func streamHook(hook string, rd io.ReadCloser) error {
	defer rc.Close()
	scanner := bufio.NewScanner(rc)
	log := log.WithFields(log.Fields{trace.Component: "hook", "hook": hook})
	for scanner.Scan() {
		log.Info(scanner.Text())
	}
	if err := scanner.Err(); err != nil {
		return trace.Wrap(err)
	}
	return nil
}
```

File: tool/gravity/cli/update.go [hunk 3]
```diff
@@ -51,7 +52,14 @@ func updateTrigger(
 	localEnv *localenv.LocalEnvironment,
 	upgradeEnv *localenv.LocalEnvironment,
 	appPackage string,
-	manual bool) (err error) {
+	manual bool,
+	systemLogFile string,
+) error {
+	err := install.InitLogging(systemLogFile)
```
Comment (a-palchikov) [inline tool/gravity/cli/update.go]: This could be done outside - prior to calling the function. Same for `rpcAgentRun`

**Comment by knisbet** [comment]
By moving the log, I'm curious about the implications for existing users. Have you considered symlinking where logs from their existing location to the /var/log directory. Or dropping a local file, that indicates the new location? It's just one of those small things, where I think lots of our users have muscle memory about where to find these logs.

Also, related to upgrades, I think there is a second issue that may not have been captured in gravitational/gravity.e#3596 or was captured somewhere else. The gravity agent that is coordinating the upgrade may be on a different node then where it was triggered. So I thought part of this issue was to try and address that, by ensuring the rpcAgent coordinating the upgrade would be on the same node that triggered it, so a complete log would be available. Just wanted to make sure that component of this problem didn't get lost somewhere.

File: lib/update/phase_app.go (hunk 2 above)
Comment (r0mant) [reply lib/update/phase_app.go]: I actually did that initially (added "hook" field) but it was always hook=update so I doubted its usefulness and removed. I can bring it back though.

File: lib/defaults/defaults.go [hunk 4]
```diff
@@ -211,10 +211,10 @@ const (
 	GravityPackagePrefix = "gravitational.io/gravity"
 
 	// TelekubeSystemLog defines the default location for the system log
-	TelekubeSystemLog = "./telekube-system.log"
+	TelekubeSystemLog = "/var/log/telekube-system.log"
```
Comment (klizhentas) [inline lib/defaults/defaults.go]: why this change?

Comment (r0mant) [reply lib/defaults/defaults.go]: I explained it in the PR description.

> This gives us a stable well-known location where these files can be found, as opposed to currently having to search for them everywhere because the working dir may be completely different depending on how the command was run (e.g. if it's run via an agent, then the file will end up in / and so on).

Comment (klizhentas) [reply lib/defaults/defaults.go]: ok, makes sense

**Comment by r0mant** [comment]
@knisbet Honestly, I'd be surprised if anyone had muscle memory of finding this log file because I can't find it myself a lot of times :) I guess I could still write to /var/log and do a symlink from cwd to it. I think it was an unfortunate design decision to put it in cwd to start with. We end up with garbage files laying around which even gravity leave can't cleanup, which was also part of the reason I did this (I already moved rpcsecrets and other .db files we used to create in cwd elsewhere).

I will see if it's easy to make sure master agent is started on the same node and if so will implement. Otherwise, will make a ticket to address separately.

**Comment by knisbet** [comment]
I agree /var/log is a better location for it, it's just I think from what I see at customers, is they do go to this log when an installation fails. So, all I'm getting at is moving it while better, we just need to be careful we're not making confusing changes.

File: tool/gravity/cli/update.go [hunk 5]
```diff
@@ -132,6 +125,13 @@ func updateTrigger(
 
 	if !manual {
 		req.leaderParams = []string{constants.RpcAgentUpgradeFunction}
+		// attempt to schedule the master agent on this node but do not
+		// treat the failure to do so as critical
+		req.leader, err = findLocalServer(*cluster)
+		if err != nil {
+			log.Errorf("Failed to determine local node: %v.",
```
Comment (a-palchikov) [inline tool/gravity/cli/update.go]: `log.Warnf` if it's not critical

Comment (r0mant) [reply tool/gravity/cli/update.go]: Will change in the next PR so I don't have to wait for CI again.
